worker

```diff
@@ -11,62 +52,96 @@ BH_UUID=${BH_UUID}
 EOF
 )
 
-GITEA_REPO="$GITEA_REPOSITORY_ROOT"/"${BH_USER}"/"${BH_REPO}".git
-#echo $GITEA_REPO #DEBUG
+# Canonicalize the GITEA_REPOSITORY_ROOT before changing directories
+# in case it was a relative path.
+GITEA_REPO=$(realpath --canonicalize-existing \
+  "${GITEA_REPOSITORY_ROOT}/${BH_USER}/${BH_REPO}.git")
```

Comment:
I see. I actually ran my tests with relative paths, and everything failed, that's why I put this in. And since I wasn't trying to show this path to users, only in the admin logs, I hadn't thought about showing a pretty path. Good point.